src/aal/aal_arm.h

```diff
@@ -49,7 +49,9 @@ namespace snmalloc
 #ifdef _MSC_VER
       __prefetch(ptr);
 #else
-#  ifdef SNMALLOC_VA_BITS_64
+#  if defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
Is there a reason not to just use __builtin_prefetch for AArch64 and Morello? GCC 5.4 and Clang 9.0.0 both map it to prfm pldl1keep, which are the oldest godbolt.org has. 32-bit Arm is sadder, Clang still doesn't map it to anything and GCC looks like it might be 9.3+.